sensuflow.sh

```diff
@@ -98,12 +103,15 @@ if [ $status -lt 200 ] || [ $status -ge 400 ]; then
 	echo "Probe of "$SENSU_BACKEND_URL/health" returned status code: $status"
 	exit 1
 fi
-if [[ $VERBOSE ]]; then echo "Checking Sensu auth credentials"; fi
-status=$(${curl_command} --connect-timeout 30 -k -o /dev/null -s -w "%{http_code}" -X GET "${SENSU_BACKEND_URL}/auth/test" -u "${SENSU_USER}:${SENSU_PASSWORD}")
-if [ $status -lt 200 ] || [ $status -ge 400 ]; then
-	echo "Sensu auth failed"
-	echo "Probe of "$SENSU_BACKEND_URL/auth/test" returned status code: $status"
-	exit 1
+
+if [ -z "$SENSU_API_KEY" ]; then # Skip auth test if API KEY is specified
+  if [[ $VERBOSE ]]; then echo "Checking Sensu auth credentials"; fi
+  status=$(${curl_command} --connect-timeout 30 -k -o /dev/null -s -w "%{http_code}" -X GET "${SENSU_BACKEND_URL}/auth/test" -u "${SENSU_USER}:${SENSU_PASSWORD}")
```

Comment:
This test doesn't have to be skipped, it just needs to be modified to pass the api key as an http header.

Reply:
Nope I'm wrong, the auth/test endpoint is only valid for user/pass.
Okay this looks good as it is. I'm going to do a follow up pr to refactor the github action definition that uses this script to allow for api key use